src/Workspaces/Core/Portable/FindSymbols/FindReferences/FindReferencesSearchEngine.cs

```diff
@@ -21,39 +21,26 @@ namespace Microsoft.CodeAnalysis.FindSymbols;
 
 using Reference = (SymbolGroup group, ISymbol symbol, ReferenceLocation location);
 
-internal partial class FindReferencesSearchEngine
+internal partial class FindReferencesSearchEngine(
+    Solution solution,
+    IImmutableSet<Document>? documents,
+    ImmutableArray<IReferenceFinder> finders,
+    IStreamingFindReferencesProgress progress,
+    FindReferencesSearchOptions options)
 {
-    private readonly Solution _solution;
-    private readonly IImmutableSet<Document>? _documents;
-    private readonly ImmutableArray<IReferenceFinder> _finders;
-    private readonly IStreamingProgressTracker _progressTracker;
-    private readonly IStreamingFindReferencesProgress _progress;
-    private readonly FindReferencesSearchOptions _options;
-
-    private static readonly TaskScheduler s_exclusiveScheduler = new ConcurrentExclusiveSchedulerPair().ExclusiveScheduler;
-
     /// <summary>
-    /// Mapping from symbols (unified across metadata/retargeting) and the set of symbols that was produced for 
-    /// them in the case of linked files across projects.  This allows references to be found to any of the unified
-    /// symbols, while the user only gets a single reported group back that corresponds to that entire set.
+    /// Scheduler we use when we're doing operations in the BG and we want to rate limit them to not saturate the threadpool.
     /// </summary>
-    private readonly ConcurrentDictionary<ISymbol, SymbolGroup> _symbolToGroup = new(MetadataUnifyingEquivalenceComparer.Instance);
```

Comment:
removed this from being ambient state to just being a normal scratch buffer created within the Far algorithm (like all the other maps/sets/lists it has.